staging/src/k8s.io/metrics/hack/update-codegen.sh

```diff
@@ -28,13 +28,13 @@ CODEGEN_PKG=${CODEGEN_PKG:-$(cd ${SCRIPT_ROOT}; ls -d -1 ./vendor/k8s.io/code-ge
 #
 # we skip informers and listers for metrics, because we don't quite support the requisite operations yet
 # we skip generating the internal clientset as it's not really needed
-${CODEGEN_PKG}/generate-internal-groups.sh deepcopy,conversion \
+"${CODEGEN_PKG}/generate-internal-groups.sh" deepcopy,conversion \
   k8s.io/metrics/pkg/client k8s.io/metrics/pkg/apis k8s.io/metrics/pkg/apis \
   "metrics:v1alpha1,v1beta1 custom_metrics:v1beta1 external_metrics:v1beta1" \
-  --output-base "$(dirname ${BASH_SOURCE})/../../.." \
-  --go-header-file ${SCRIPT_ROOT}/hack/boilerplate.go.txt
-${CODEGEN_PKG}/generate-groups.sh client \
+  --output-base "$(dirname "${BASH_SOURCE[0]}")/../../.." \
```

Comment:
I would pull this out into its own OUTPUT_BASE var above, so you don't have the weird nested quotes thing going on here, ie:
```sh
OUTPUT_BASE=$(dirname "${BASH_SOURCE[0]}")/../../..
#...
--output-base "${OUTPUT_BASE}"
```

You might also consider using `SCRIPT_ROOT` instead of redoing the `BASH_SOURCE` thing.

Reply:
Brother, this modification will cause additional problems.
It will look like this:
```
In ./staging/src/k8s.io/metrics/hack/update-codegen.sh line 35:
  --output-base "${OUTPUT_BASE}"
  ^-----------^ SC2215: This flag is used as a command name. Bad line break or missing [ .. ]?
In ./staging/src/k8s.io/metrics/hack/update-codegen.sh line 36:
  --go-header-file "${SCRIPT_ROOT}/hack/boilerplate.go.txt"
  ^--------------^ SC2215: This flag is used as a command name. Bad line break or missing [ .. ]?
```

Reply:
You are likely missing `\` at the end of the line above it and on the end of the line containing `--output-base "${OUTPUT_BASE}"`

It should look similar to this:
```
"${CODEGEN_PKG}/generate-internal-groups.sh" deepcopy,conversion \
  k8s.io/metrics/pkg/client k8s.io/metrics/pkg/apis k8s.io/metrics/pkg/apis \
  "metrics:v1alpha1,v1beta1 custom_metrics:v1beta1 external_metrics:v1beta1" \ 
  --output-base "${OUTPUT_BASE}" \
  --go-header-file "${SCRIPT_ROOT}/hack/boilerplate.go.txt"
```

Reply:
```
--output-base "${OUTPUT_BASE}" \
```
Appear above problem.

Reply:
I agree with @mrbobbytables the error really reads like a missing trailing `\`, shellcheck was happy with my suggested changes locally

Since I can't see the source you're running this against, my advice would be to look _above_ your `output-base` and `go-header-file` lines